Framework/DataHandling/src/LoadRawHelper.cpp

```diff
@@ -6,36 +6,94 @@
 // SPDX - License - Identifier: GPL - 3.0 +
 #include "MantidDataHandling/LoadRawHelper.h"
 #include "LoadRaw/isisraw2.h"
+
 #include "MantidAPI/Axis.h"
 #include "MantidAPI/FileProperty.h"
 #include "MantidAPI/SpectrumDetectorMapping.h"
 #include "MantidAPI/WorkspaceFactory.h"
 #include "MantidAPI/WorkspaceGroup.h"
-#include "MantidDataHandling/LoadLog.h"
 #include "MantidDataHandling/RawFileInfo.h"
 #include "MantidDataObjects/Workspace2D.h"
 #include "MantidGeometry/Instrument.h"
-#include "MantidKernel/ArrayProperty.h"
-#include "MantidKernel/ConfigService.h"
 #include "MantidKernel/Glob.h"
 #include "MantidKernel/ListValidator.h"
 #include "MantidKernel/OptionalBool.h"
 #include "MantidKernel/Strings.h"
-#include "MantidKernel/TimeSeriesProperty.h"
 #include "MantidKernel/UnitFactory.h"
-
-#include <boost/algorithm/string/predicate.hpp>
-#include <boost/date_time/gregorian/gregorian.hpp>
-#include <memory>
-
-#include <Poco/DateTimeFormat.h>
-#include <Poco/DateTimeParser.h>
-#include <Poco/DirectoryIterator.h>
 #include <Poco/File.h>
 #include <Poco/Path.h>
+#include <boost/algorithm/string/predicate.hpp>
+#include <cstdio>
 
-#include <cmath>
-#include <cstdio> //Required for gcc 4.4
+namespace {
+/**
+ * Return the path to the alternate data stream for the given path
+ * @param filePath The full path to the main data path
+ */
+inline std::string alternateDataStream(const Poco::Path &filePath) {
+  return filePath.toString() + ":checksum";
+}
+
+/**
+ * This method looks for ADS with name checksum exists
+ * @param pathToFile The path and name of the file.
+ * @return True if ADS stream checksum exists
+ */
+inline bool hasAlternateDataStream(const Poco::Path &pathToFile) {
+#ifdef _WIN32
+  std::ifstream adsStream(alternateDataStream(pathToFile));
+  if (!adsStream) {
+    return false;
+  }
+  adsStream.close();
+  return true;
+#else
+  UNUSED_ARG(pathToFile);
```

Comment:
Could we add the [[[maybe_unused]] ](https://en.cppreference.com/w/cpp/language/attributes/maybe_unused) attribute to the function parameter `pathToFile`. I think it achieves the same thing, but it would avoid having to using this macro.